lib/nerves_hub_link/socket.ex

```diff
@@ -468,15 +468,16 @@ defmodule NervesHubLink.Socket do
 
   @impl Slipstream
   def handle_info(:connect_check_network_availability, socket) do
-    hostname = URI.parse(socket.assigns.config.socket[:url]).host
+    uri = URI.parse(socket.assigns.config.socket[:url])
 
-    case :inet.gethostbyname(to_charlist(hostname)) do
-      {:ok, _} ->
+    case :gen_tcp.connect(to_charlist(uri.host), uri.port, [active: false, packet: 0], 2_000) do
+      {:ok, tcp_socket} ->
+        :gen_tcp.close(tcp_socket)
```

Comment:
Any considerations needed since we're opening and closing a socket now? The previous implementation just resolved the hostname.